File: java/src/test/java/problems1/ZigzagConversionTest.java
package problems1;

import org.junit.Test;
import testing.Sample;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ZigzagConversionTest extends SampledTest {
    @Override
    public String getFileName() {
        return "zigzag_conversion.json";
    }

    @Test
    public void test() {
        for (ZigzagConversion solution : ZigzagConversion.values()) {
            for (TypedSample sample : getSamples(TypedSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(
                        solution.convert(
                            sample.input.s,
                            sample.input.numRows
                        )
                    ).isEqualTo(sample.output);
            }
        }
    }

    public static class TypedSample extends Sample<TypedSample.Input, String> {
        public static class Input {
            public String s;
            public int numRows;
        }
    }
}

File: typescript/__tests__/problems1_100/zigzag-conversion.test.ts
import {strictEqual} from 'assert';
import SOLUTIONS from '../../src/problems1_100/zigzag-conversion';
import getSamples from '../../../testing/js/src/sample/samples';

type Sample = {
  input: {
    s: string,
    numRows: number,
  },
  output: string,
  message: string,
}

describe(
    'ZigzagConversion',
    () =>
        test(
            'test',
            () =>
                SOLUTIONS.forEach(solution => {
                  getSamples().forEach((sample: Sample) => {
                    strictEqual(
                        solution.convert(
                            sample.input.s,
                            sample.input.numRows,
                        ),
                        sample.output,
                        sample.message,
                    );
                  });
                }),
        ),
);
